### ISSUE TITLE
Firebase Unity SDK - Unity Editor crashed after second play - Auth

### ISSUE THREAD

**Issue by dzennik** [firebase/quickstart-unity#720]
### Please fill in the following fields:
Unity editor version: tested on (Unity 2019.4.1f1, Unity 2019.3.9f1, Unity 2019.2.21f1)
Firebase Unity SDK version: 6.15.0 (Don't reproduce on the 6.14.1)
Source you installed the SDK (.unitypackage or Unity Package Manager): (Unity Package Manager)
Firebase plugins in use (Auth, Database, etc.): Auth
Additional SDKs you are using (Facebook, AdMob, etc.):  None
Platform you are using the Unity editor on (Mac, Windows, or Linux):  Mac, Unity Editor
Platform you are targeting (iOS, Android, and/or desktop): iOS
Scripting Runtime (Mono, and/or IL2CPP): IL2CPP

### Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)

```
using Firebase;
using Firebase.Auth;
using UnityEngine;

public class TestScript : MonoBehaviour
{
    protected FirebaseAuth auth;

    protected void Awake()
    {
        FirebaseApp.CheckAndFixDependenciesAsync().ContinueWith(task => 
        {
            var dependencyStatus = task.Result;
            if (dependencyStatus == DependencyStatus.Available) 
            {
                auth = FirebaseAuth.GetAuth(FirebaseApp.Create());
            }
        });
    }
}
```

The unity editor on mac is crashing after second play. The manual calling auth.Dispose() method leads to crash.

Don't reproduce on the 6.14.1 FireBase library version!!!

### Please answer the following, if applicable:
Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?

No. I don't try it. I just import Firebase Core & Auth from Unity Package Manager.

What's the issue repro rate? (eg 100%, 1/5 etc)

100%

**Comment by Barokoli** [comment]
I have the same issue with platform target Android. 6.15 crashes most of the time on second play. Sometimes I reach third play but then it crashes always.
Firebase Auth and core installed via Packagemanager. Other than that no other packages were installed.
Might also be an issue with the external dependency manager?

**Comment by a-maurice** [comment]
Hi all,

Sorry for the delay in response.  We've just released 6.15.1, which should fix this issue, if you can update to use it.

**Comment by ol1ph4nt** [comment]
Hi,

I've updated to 6.15.1 but I'm still having the unity editor crashed on second play.
In my case, I'm testing on PC, targeting Android platform.

One possible reason why it is happening:
I've narrowed down the issue to getting data from the realtime database.
In the editor, during play, whenever I'm trying to query data with added Sorting/Filtering methods (OrderBy, LimitTo), the task doesn't seem to be able to complete.
Then, after stopping play, if I do any action such as a second play or even recompile code, Unity just crashes.

Anyone encountering a similar issue?
